custom-webpack@^11 not found

### Demonstration
Executing `npm install` just after creating the Angular 11 parcel produces the error "No matching version found for @angular-builders/custom-webpack@^11".
This is using single-spa-angular@4.8.0, after the fix from #304 .

Created the project here:
https://github.com/daniloesk/single-spa-errors/tree/single-spa-angular%23306

Command logs:
```
USER:HOME$ npx create-single-spa --dir . --moduleType app-parcel --framework angular
? Project name (can use letters, numbers, dash or underscore) ng11error
? Do you want to enforce stricter type checking and stricter bundle budgets in the workspace?
  This setting helps improve maintainability and catch bugs ahead of time.
  For more information, see https://angular.io/strict Yes
? Would you like to add Angular routing? Yes
? Which stylesheet format would you like to use? CSS
CREATE ng11error/README.md (1018 bytes)
CREATE ng11error/.editorconfig (274 bytes)
CREATE ng11error/.gitignore (631 bytes)
CREATE ng11error/angular.json (3647 bytes)
CREATE ng11error/package.json (1199 bytes)
CREATE ng11error/tsconfig.json (737 bytes)
CREATE ng11error/tslint.json (3185 bytes)
CREATE ng11error/.browserslistrc (703 bytes)
CREATE ng11error/karma.conf.js (1104 bytes)
CREATE ng11error/tsconfig.app.json (287 bytes)
CREATE ng11error/tsconfig.spec.json (333 bytes)
CREATE ng11error/src/favicon.ico (948 bytes)
CREATE ng11error/src/index.html (295 bytes)
CREATE ng11error/src/main.ts (372 bytes)
CREATE ng11error/src/polyfills.ts (2826 bytes)
CREATE ng11error/src/styles.css (80 bytes)
CREATE ng11error/src/test.ts (753 bytes)
CREATE ng11error/src/assets/.gitkeep (0 bytes)
CREATE ng11error/src/environments/environment.prod.ts (51 bytes)
CREATE ng11error/src/environments/environment.ts (662 bytes)
CREATE ng11error/src/app/app-routing.module.ts (245 bytes)
CREATE ng11error/src/app/app.module.ts (393 bytes)
CREATE ng11error/src/app/app.component.css (0 bytes)
CREATE ng11error/src/app/app.component.html (25757 bytes)
CREATE ng11error/src/app/app.component.spec.ts (1066 bytes)
CREATE ng11error/src/app/app.component.ts (213 bytes)
CREATE ng11error/e2e/protractor.conf.js (904 bytes)
CREATE ng11error/e2e/tsconfig.json (274 bytes)
CREATE ng11error/e2e/src/app.e2e-spec.ts (660 bytes)
CREATE ng11error/e2e/src/app.po.ts (274 bytes)
✔ Packages installed successfully.
    Successfully initialized git.
Installing packages for tooling via npm.
Installed packages for tooling via npm.
? Does your application use Angular routing? Yes
? Does your application use BrowserAnimationsModule? No
    Added 'single-spa' as a dependency
    Added 'single-spa-angular' as a dependency
    Added '@angular-builders/custom-webpack' as a dependency
    Generated 'main.single-spa.ts
    Generated 'single-spa-props.ts
    Generated asset-url.ts
    Generated extra-webpack.config.js
    Using @angular-devkit/custom-webpack builder.
    Updated angular.json configuration
    @angular-builders/custom-webpack:browser
CREATE extra-webpack.config.js (303 bytes)
CREATE src/main.single-spa.ts (933 bytes)
CREATE src/app/empty-route/empty-route.component.ts (143 bytes)
CREATE src/single-spa/asset-url.ts (502 bytes)
CREATE src/single-spa/single-spa-props.ts (333 bytes)
UPDATE package.json (1560 bytes)
UPDATE tsconfig.app.json (188 bytes)
UPDATE angular.json (3844 bytes)
For further routing setup, see https://single-spa.js.org/docs/ecosystem-angular#configure-routes
Project setup complete!
Steps to test your Angular single-spa application:
1. Run 'npm run serve:single-spa:ng11error'
2. Go to http://single-spa-playground.org/playground/instant-test?name=ng11error&url=%2F%2Flocalhost%3A4200%2Fmain.js&framework=angular to see it working!
```

```
USER:single-spa-angular-11-error$ npm install
npm ERR! code ETARGET
npm ERR! notarget No matching version found for @angular-builders/custom-webpack@^11.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
npm ERR! notarget 
npm ERR! notarget It was specified as a dependency of 'single-spa-angular-11-error'
npm ERR! notarget 

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/USER/.npm/_logs/2020-11-23T17_21_10_478Z-debug.log
```

### Expected Behavior
`npm install` completes successfully.

### Actual Behavior
Error "No matching version found for @angular-builders/custom-webpack@^11".

Got a strange error after using custom-webpack@latest, but I think it is unrelated. Compared with another generated project and it is different, so I will recreate the error project from scratch.
```
An unhandled exception occurred: tsconfig.app.json:2:3 - error TS1327: String literal with double quotes expected.
```

Using custom-webpack@latest did not work. The project does install, but running it gives the error below. Building gives the same error message. Maybe this was not meant to work after all (different version).

```
USER:single-spa-angular-11-error$ npm run serve:single-spa:ng11error

> single-spa-angular-11-error@0.0.0 serve:single-spa:ng11error /home/USER/single-spa-angular-11-error
> ng s --project ng11error --disable-host-check --port 4200 --deploy-url http://localhost:4200/ --live-reload false

WARNING: Running a server with --disable-host-check is a security risk. See https://medium.com/webpack/webpack-dev-server-middleware-security-issues-1489d950874a for more information.
WARNING: --deploy-url and/or --base-href contain unsupported values for ng serve. Default serve path of '/' used. Use --serve-path to override.
An unhandled exception occurred: webpackMerge.smart is not a function
See "/tmp/ng-ZvEg4Z/angular-errors.log" for further details.
npm ERR! code ELIFECYCLE
npm ERR! syscall spawn
npm ERR! file sh
npm ERR! errno ENOENT
npm ERR! single-spa-angular-11-error@0.0.0 serve:single-spa:ng11error: `ng s --project ng11error --disable-host-check --port 4200 --deploy-url http://localhost:4200/ --live-reload false`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the single-spa-angular-11-error@0.0.0 serve:single-spa:ng11error script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/USER/.npm/_logs/2020-11-24T11_56_35_646Z-debug.log
```

Log:
```
0 info it worked if it ends with ok
1 verbose cli [
1 verbose cli   '/opt/nvm/versions/node/v14.15.1/bin/node',
1 verbose cli   '/opt/nvm/versions/node/v14.15.1/bin/npm',
1 verbose cli   'run',
1 verbose cli   'serve:single-spa:ng11error'
1 verbose cli ]
2 info using npm@6.14.8
3 info using node@v14.15.1
4 verbose run-script [
4 verbose run-script   'preserve:single-spa:ng11error',
4 verbose run-script   'serve:single-spa:ng11error',
4 verbose run-script   'postserve:single-spa:ng11error'
4 verbose run-script ]
5 info lifecycle single-spa-angular-11-error@0.0.0~preserve:single-spa:ng11error: single-spa-angular-11-error@0.0.0
6 info lifecycle single-spa-angular-11-error@0.0.0~serve:single-spa:ng11error: single-spa-angular-11-error@0.0.0
7 verbose lifecycle single-spa-angular-11-error@0.0.0~serve:single-spa:ng11error: unsafe-perm in lifecycle true
8 verbose lifecycle single-spa-angular-11-error@0.0.0~serve:single-spa:ng11error: PATH: /opt/nvm/versions/node/v14.15.1/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/USER/single-spa-angular-11-error/node_modules/.bin:/opt/nvm/versions/node/v14.15.1/bin:/home/USER/.local/bin:/home/USER/bin:/home/USER/.local/bin:/home/USER/bin:/opt/nvm/versions/node/v14.15.1/bin:/home/USER/.local/bin:/home/USER/bin:/home/USER/.local/bin:/home/USER/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/opt/apache/maven/current/bin:/opt/nexus/repository/current/bin:/opt/android/sdk/emulator:/opt/android/studio/current/bin:/opt/apache/maven/current/bin:/opt/nexus/repository/current/bin:/opt/apache/maven/current/bin:/opt/nexus/repository/current/bin:/opt/android/sdk/emulator:/opt/android/studio/current/bin:/opt/apache/maven/current/bin:/opt/nexus/repository/current/bin
9 verbose lifecycle single-spa-angular-11-error@0.0.0~serve:single-spa:ng11error: CWD: /home/USER/single-spa-angular-11-error
10 silly lifecycle single-spa-angular-11-error@0.0.0~serve:single-spa:ng11error: Args: [
10 silly lifecycle   '-c',
10 silly lifecycle   'ng s --project ng11error --disable-host-check --port 4200 --deploy-url http://localhost:4200/ --live-reload false'
10 silly lifecycle ]
11 info lifecycle single-spa-angular-11-error@0.0.0~serve:single-spa:ng11error: Failed to exec serve:single-spa:ng11error script
12 verbose stack Error: single-spa-angular-11-error@0.0.0 serve:single-spa:ng11error: `ng s --project ng11error --disable-host-check --port 4200 --deploy-url http://localhost:4200/ --live-reload false`
12 verbose stack spawn ENOENT
12 verbose stack     at ChildProcess.<anonymous> (/opt/nvm/versions/node/v14.15.1/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:48:18)
12 verbose stack     at ChildProcess.emit (events.js:315:20)
12 verbose stack     at maybeClose (internal/child_process.js:1048:16)
12 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5)
13 verbose pkgid single-spa-angular-11-error@0.0.0
14 verbose cwd /home/USER/single-spa-angular-11-error
15 verbose Linux 4.18.0-193.28.1.el8_2.x86_64
16 verbose argv "/opt/nvm/versions/node/v14.15.1/bin/node" "/opt/nvm/versions/node/v14.15.1/bin/npm" "run" "serve:single-spa:ng11error"
17 verbose node v14.15.1
18 verbose npm  v6.14.8
19 error code ELIFECYCLE
20 error syscall spawn
21 error file sh
22 error errno ENOENT
23 error single-spa-angular-11-error@0.0.0 serve:single-spa:ng11error: `ng s --project ng11error --disable-host-check --port 4200 --deploy-url http://localhost:4200/ --live-reload false`
23 error spawn ENOENT
24 error Failed at the single-spa-angular-11-error@0.0.0 serve:single-spa:ng11error script.
24 error This is probably not a problem with npm. There is likely additional logging output above.
25 verbose exit [ 1, true ]
```

--- a/schematics/ng-add/dependencies.ts
+++ b/schematics/ng-add/dependencies.ts
@@ -1,3 +1,6 @@
+import * as https from 'https';
+import { IncomingMessage } from 'http';
+import { VERSION } from '@angular/core';
 import { NodeDependencyType, NodeDependency } from '@schematics/angular/utility/dependencies';
 
 interface PackageJson {
@@ -33,14 +36,61 @@ export function getSingleSpaAngularDependency(): NodeDependency {
   };
 }
 
-export function getAngularBuildersCustomWebpackDependency(): NodeDependency {
+export async function getAngularBuildersCustomWebpackDependency(): Promise<NodeDependency> {
   return {
     name: '@angular-builders/custom-webpack',
-    // Relying on the current major version of Angular is wrong.
-    // We have to install the `latest` version because we cannot know
-    // when the new version will be published.
-    version: 'latest',
     overwrite: false,
     type: NodeDependencyType.Dev,
+    version: await resolveCustomWebpackVersion(),
   };
+}
+
+async function resolveCustomWebpackVersion(): Promise<string> {
+  let version: string;
+
+  try {
+    const versions: string[] = await getCustomWebpackVersions();
+    // We do `filter` because there can be `beta` versions, thus `^11`
+    // will not work in that case.
+    const compatibleMajorVersions: string[] = versions.filter(version =>
+      version.startsWith(VERSION.major),
+    );
+
+    version = compatibleMajorVersions.pop() || 'latest';
+  } catch {
+    // We could actually initialize version with the `latest` value,
+    // but let's be more imperative and fallback to the `latest` value
+    // if any exception has occured previously.
+    version = 'latest';
+  }
+
+  return version;
+}
+
+function getCustomWebpackVersions(): Promise<string[]> {
+  return new Promise((resolve, reject) => {
+    const request = https.get(
+      {
+        protocol: 'https:',
+        hostname: 'registry.npmjs.com',
+        path: '/@angular-builders/custom-webpack',
+      },
+      (response: IncomingMessage) => {
+        const chunks: Buffer[] = [];
+
+        response
+          .on('error', reject)
+          .on('data', chunk => {
+            chunks.push(chunk);
+          })
+          .on('end', () => {
+            const response = JSON.parse(`${Buffer.concat(chunks)}`);
+            const versions: string[] = Object.keys(response.versions);
+            resolve(versions);
+          });
+      },
+    );
+
+    request.on('error', reject).end();
+  });
 }

--- a/schematics/ng-add/dependencies.ts
+++ b/schematics/ng-add/dependencies.ts
@@ -33,16 +33,13 @@ export function getSingleSpaAngularDependency(): NodeDependency {
   };
 }
 
-/**
- * We have to install `@angular-builders/custom-webpack` version compatible with the current
- * version of Angular. If Angular is 8 then `custom-webpack@8.4.1` has to be installed.
- */
 export function getAngularBuildersCustomWebpackDependency(): NodeDependency {
-  const { VERSION } = require('@angular/core');
-
   return {
     name: '@angular-builders/custom-webpack',
-    version: `^${VERSION.major}`,
+    // Relying on the current major version of Angular is wrong.
+    // We have to install the `latest` version because we cannot know
+    // when the new version will be published.
+    version: 'latest',
     overwrite: false,
     type: NodeDependencyType.Dev,
   };

--- a/schematics/ng-add/index.ts
+++ b/schematics/ng-add/index.ts
@@ -51,14 +51,14 @@ export default function (options: NgAddOptions): Rule {
 }
 
 export function addDependencies(): Rule {
-  const dependencies: NodeDependency[] = [
+  const dependencies: Array<NodeDependency | Promise<NodeDependency>> = [
     getSingleSpaDependency(),
     getSingleSpaAngularDependency(),
     getAngularBuildersCustomWebpackDependency(),
   ];
 
-  return (tree: Tree, context: SchematicContext) => {
-    for (const dependency of dependencies) {
+  return async (tree: Tree, context: SchematicContext) => {
+    for await (const dependency of dependencies) {
       addPackageJsonDependency(tree, dependency);
       context.logger.info(`Added '${dependency.name}' as a dependency`);
     }

